File: 조준장/level2/시소 짝꿍.py
from collections import defaultdict


def solution(weights):
    answer = 0
    seesaw = defaultdict(int)

    for w in weights:
        answer += seesaw[w]
        answer += seesaw[(w * 2) / 3] 
        answer += seesaw[(w * 3) / 2]
        answer += seesaw[(w * 3) / 4] 
        answer += seesaw[(w * 4) / 3]
        answer += seesaw[(w * 4) / 2] 
        answer += seesaw[(w * 2) / 4]
        seesaw[w] += 1

    return answer

File: 이훈일/Programmers/Level2/시소 짝꿍.java
import java.util.*;

class Solution {
    public long solution(int[] weights) {
        Arrays.sort(weights);
        
        int[] count = new int[weights[weights.length-1] * 4 + 1];
        long answer = 0;
        
        for(int i = 0; i < weights.length; i++) {
            int weight = weights[i];
            
            if(count[weight * 4] > 0) {
                answer += count[weight * 4];
            } 
                
            for(int seat = 2; seat <= 4; seat++) {
                answer += count[weight * seat];
            }
            
            answer -= count[weight * 4] * 3;
            
            for(int seat = 2; seat <= 4; seat++) {
                count[weight * seat] += 1;
            }
        }
        
        return answer;
    }
}
